lib/muser.h

```diff
@@ -165,6 +165,45 @@ enum {
     LM_DEV_NUM_REGS = 9
 };
 
+/**
+ * Callback function that gets called when a capability is accessed. The
+ * callback is not called when the ID and next fields are accessed, these are
+ * handled by the library.
+ *
+ * @pvt: private pointer
+ * @id: capability ID being accessed
+ * @buf: pointer to data being read or written
+ * @count: number of bytes being read or written
+ * @offset: offset within the capability
+ * @is_write: whether the capability is read or written
+ *
+ * @returns the number of bytes read or written
+ */
+typedef ssize_t (lm_cap_access_t) (void *pvt, uint8_t id,
+                                   char *buf, size_t count,
+                                   loff_t offset, bool is_write);
+
+typedef struct {
+
+    /*
+     * Capability ID, as defined by the PCI specification. Also defined as
+     * PCI_CAP_ID_XXX in <linux/pci_regs.h>.
+     */
+    uint8_t id;
+
+    /*
+     * Size of the capability.
+     */
+    size_t size;
+
+    /*
+     * Function to call back when the capability gets read or written.
+     */
+    lm_cap_access_t *fn;
+} lm_cap_t;
+
+#define LM_MAX_CAPS (PCI_CFG_SPACE_SIZE - PCI_STD_HEADER_SIZEOF) / PCI_CAP_SIZEOF
```

Comment:
Math in defines (the whole thing) should always be within brackets.
(I can see this was already wrong, but this is a good opportunity to fix it.)